toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -52,20 +55,59 @@ func NewIsoMaker(unattendedInstall bool, baseDirPath, buildDirPath, releaseVersi
 	if baseDirPath == "" {
 		baseDirPath = filepath.Dir(configFilePath)
 	}
+
+	imageNameBase := strings.TrimSuffix(filepath.Base(configFilePath), ".json")
+
 	if imageNameTag != "" {
 		imageNameTag = "-" + imageNameTag
 	}
 
+	// readConfigFile() and verifyConfig() panic if an error occurs.
+	config := readConfigFile(configFilePath, baseDirPath)
+	verifyConfig(config, unattendedInstall)
+
 	return &IsoMaker{
+		enableBiosBoot:     true,
 		unattendedInstall:  unattendedInstall,
+		config:             config,
 		baseDirPath:        baseDirPath,
 		buildDirPath:       buildDirPath,
 		initrdPath:         initrdPath,
 		releaseVersion:     releaseVersion,
 		resourcesDirPath:   resourcesDirPath,
-		configFilePath:     configFilePath,
 		fetchedRepoDirPath: isoRepoDirPath,
 		outputDirPath:      outputDir,
+		imageNameBase:      imageNameBase,
+		imageNameTag:       imageNameTag,
+	}
+}
+
+func NewIsoMakerWithConfig(unattendedInstall, enableBiosBoot bool, baseDirPath, buildDirPath, releaseVersion, resourcesDirPath string, config configuration.Config, initrdPath, grubCfgPath, isoRepoDirPath, outputDir, imageNameBase, imageNameTag string) *IsoMaker {
+
+	if imageNameBase == "" {
+		imageNameBase = defaultImageNameBase
+	}
+
+	if imageNameTag != "" {
+		imageNameTag = "-" + imageNameTag
```

Comment:
What if `imageNameTag` already begins with a hyphen? Do we want it doubled?

Reply:
I got to the changes inside `buildIsoImageFilePath`. This change seems to be related to that one. I'd suggest to move this logic into `buildIsoImageFilePath` completely, so it's all in one place and it's easier to understand why we modify the tag name. It's a bit cryptic here.